### ISSUE TITLE
HTML Editor - Stack Overflow When Refactoring

### ISSUE THREAD

**Issue by aksarben** [apache/netbeans#4844]
### Apache NetBeans version

Apache NetBeans 15

### What happened

The error message I got when trying to refactor/rename an HTML file.

![image](https://user-images.githubusercontent.com/6260602/197357408-16ed76c8-9e47-45ea-9c69-a132410212e0.png)
[messages.log](https://github.com/apache/netbeans/files/9845114/messages.log)

### How to reproduce

In file **aadraclo.htm**, I created a link pointing to file **link-test.htm**. Both files are in the same folder:

![image](https://user-images.githubusercontent.com/6260602/197357551-6d6c40fa-0fe6-4817-a74c-01e143bdb6fd.png)

The link code:

`<a href="link-test.htm">[This phrase](https://github.com/apache/netbeans/issues/link-test.htm)</a> is a link to another page`

I tested the link in a browser, and it worked as expected.

Then, in the project folder tree, I right-clicked the file **link-test.htm** and  chose **Refactor>Rename** from the popup context menu.

When the **Rename** dialog opened, I changed the file name to **link-tester.htm** and clicked the **Refactor** button.

After a pause of about 3 minutes, the error dialog appeared, reporting a stack overflow (see the **What happened** field above).

I Googled how to refactor file names in NetBeans, but all the examples I found were for refactoring Java code, not file names.

I need to migrate my Web site from Expression Web 4 (now obsolete & unsupported), and trying to determine whether NetBeans would be a good successor. Does NetBeans not fully support file name refactoring?

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 10 Home, version 21H2, OS build 19044.2130, Windows Feature Experience Pack 120.2212.4180.0

### JDK

java version "18.0.2" 2022-07-19 Java(TM) SE Runtime Environment (build 18.0.2+9-61) Java HotSpot(TM) 64-Bit Server VM (build 18.0.2+9-61, mixed mode, sharing)

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

Tried the file rename twice, and got the stack overflow error both times.

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by Chris2011** [comment]
If you just want to rename your file just select the file in the project/files window and hit F2. But I think it will not refactor so if your file is somehow linked somewhere else, it will not work AFAIK.

**Comment by aksarben** [comment]
The context menu clearly shows NetBeans offering to refactor the file name. When I right-click the file name, the menu option is **Refactor-Rename**. When I choose that option, a dialog opens with a field to enter the new name, and a check box that says **Rename Without Refactoring**. So clearly, it's designed to allow refactoring.

The problem is not lack of functionality in the design. The problem is that a stack overflow occurs when I attempt the rename.

**Comment by Chris2011** [comment]
Yes I know. I just gave you another opportunity if you really just want to rename this one file. It will not solve the problem anyway.
